File: go-kata/interfaces/interfaces.go
package interfaces

// List interface for all list implementations
type List[T any] interface {
	Length() int
	RemoveAt(index int) *T
	Remove(item T) *T
	Get(index int) *T
	Prepend(item T)
	Append(item T)
	InsertAt(item T, idx int)
}

File: python-kata/kata/interfaces.py
from abc import ABC, abstractmethod
from typing import TypeVar, Generic, Optional

T = TypeVar('T')

class List(ABC, Generic[T]):
    @property
    @abstractmethod
    def length(self) -> int:
        pass
    
    @abstractmethod
    def remove_at(self, index: int) -> Optional[T]:
        pass
    
    @abstractmethod
    def remove(self, item: T) -> Optional[T]:
        pass
    
    @abstractmethod
    def get(self, index: int) -> Optional[T]:
        pass
    
    @abstractmethod
    def prepend(self, item: T) -> None:
        pass
    
    @abstractmethod
    def append(self, item: T) -> None:
        pass
    
    @abstractmethod
    def insert_at(self, item: T, idx: int) -> None:
        pass